src/Domain/LeanCode.DomainModels.MassTransitRelay/InjectLifetimeScopeFilter.cs

```diff
@@ -0,0 +1,38 @@
+using GreenPipes;
+using MassTransit;
+using MassTransit.Registration;
+
+namespace LeanCode.DomainModels.MassTransitRelay
+{
+    // When combining Autofac and Microsoft.Extensions.Dependency injection
+    // some DI components are not injected correctly into the message payload,
+    // so we have to fix it by ourselves.
+    public class InjectLifetimeScopeFilter<T> : IFilter<ConsumeContext<T>>
```

Comment:
Tests, maybe?

Reply:
Do you mean a unit test for this particular class? I can add it although the only think it'll test it's wether I understood MT code correctly. There is a larger test [here](https://github.com/leancodepl/corelibrary/pull/343/files#diff-396609625e6c0a2ebaa2d90da2cc9ffdbb02128b67f5a3496b93242acc6b2c87R54), checking if everything works as expected.

Reply:
I mean unit-test style tests. They will make it easier for future devs to understand your assumptions about MT. :)